### ISSUE TITLE
Need advice on geth command for starting mining.

### ISSUE THREAD

**Issue by zydjohnHotmail** [marcosmartinez7/geth-regtest#3]
Hello:
I have installed a docker image and git the geth and git your repo, and I run the following command, you can see the output:
root@47e2f8084004:~# geth --networkid 33 --mine --miner.threads 3 --datadir=/app/node1 --nodiscover --http --http.corsdomain "*" --nat "any" --unlock 'd025f1e3f633eb4816661c68155b6bf9cc43be16' --password /app/node1/password.txt --allow-insecure-unlock
INFO [03-15|21:24:33.488] Maximum peer count                       ETH=50 LES=0 total=50
INFO [03-15|21:24:33.490] Smartcard socket not found, disabling    err="stat /run/pcscd/pcscd.comm: no such file or directory"
INFO [03-15|21:24:33.495] Set global gas cap                       cap=50,000,000
INFO [03-15|21:24:33.496] Allocated trie memory caches             clean=154.00MiB dirty=256.00MiB
INFO [03-15|21:24:33.496] Using leveldb as the backing database
INFO [03-15|21:24:33.496] Allocated cache and file handles         database=/app/node1/geth/chaindata cache=512.00MiB handles=524,288
INFO [03-15|21:24:33.522] Using LevelDB as the backing database
INFO [03-15|21:24:33.541] Opened ancient database                  database=/app/node1/geth/chaindata/ancient/chain readonly=false
INFO [03-15|21:24:33.542] Disk storage enabled for ethash caches   dir=/app/node1/geth/ethash count=3
INFO [03-15|21:24:33.542] Disk storage enabled for ethash DAGs     dir=/root/.ethash          count=2
INFO [03-15|21:24:33.542] Initialising Ethereum protocol           network=33 dbversion=<nil>
INFO [03-15|21:24:33.543]
INFO [03-15|21:24:33.543] ---------------------------------------------------------------------------------------------------------------------------------------------------------
INFO [03-15|21:24:33.543] Chain ID:  33 (unknown)
INFO [03-15|21:24:33.543] Consensus: Ethash (proof-of-work)
INFO [03-15|21:24:33.543]
INFO [03-15|21:24:33.543] Pre-Merge hard forks (block based):
INFO [03-15|21:24:33.543]  - Homestead:                   #1        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/homestead.md)
INFO [03-15|21:24:33.543]  - Tangerine Whistle (EIP 150): #2        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/tangerine-whistle.md)
INFO [03-15|21:24:33.543]  - Spurious Dragon/1 (EIP 155): #3        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/spurious-dragon.md)
INFO [03-15|21:24:33.543]  - Spurious Dragon/2 (EIP 158): #3        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/spurious-dragon.md)
INFO [03-15|21:24:33.543]  - Byzantium:                   #4        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/byzantium.md)
INFO [03-15|21:24:33.543]  - Constantinople:              #5        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/constantinople.md)
INFO [03-15|21:24:33.543]  - Petersburg:                  #<nil> (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/petersburg.md)
INFO [03-15|21:24:33.543]  - Istanbul:                    #<nil> (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/istanbul.md)
INFO [03-15|21:24:33.543]  - Berlin:                      #<nil> (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/berlin.md)
INFO [03-15|21:24:33.544]  - London:                      #<nil> (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/london.md)
INFO [03-15|21:24:33.544]
INFO [03-15|21:24:33.544] The Merge is not yet available for this network!
INFO [03-15|21:24:33.544]  - Hard-fork specification: https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/paris.md
INFO [03-15|21:24:33.544]
INFO [03-15|21:24:33.544] Post-Merge hard forks (timestamp based):
INFO [03-15|21:24:33.544]
INFO [03-15|21:24:33.544] ---------------------------------------------------------------------------------------------------------------------------------------------------------
INFO [03-15|21:24:33.544]
INFO [03-15|21:24:33.544] Loaded most recent local block           number=0 hash=3936cc..c5fa46 td=1 age=3y1mo3w
WARN [03-15|21:24:33.544] Failed to load snapshot                  err="missing or corrupted snapshot"
INFO [03-15|21:24:33.544] Rebuilding state snapshot
INFO [03-15|21:24:33.545] Resuming state snapshot generation       root=b907a5..93791b accounts=0 slots=0 storage=0.00B dangling=0 elapsed="350.658µs"
INFO [03-15|21:24:33.545] Regenerated local transaction journal    transactions=0 accounts=0
INFO [03-15|21:24:33.545] Gasprice oracle is ignoring threshold set threshold=2
WARN [03-15|21:24:33.546] Error reading unclean shutdown markers   error="leveldb: not found"
WARN [03-15|21:24:33.546] Engine API enabled                       protocol=eth
WARN [03-15|21:24:33.546] Engine API started but chain not configured for merge yet
INFO [03-15|21:24:33.546] Starting peer-to-peer node               instance=Geth/v1.11.3-stable-5ed08c47/linux-amd64/go1.20.2
INFO [03-15|21:24:33.549] Generated state snapshot                 accounts=257 slots=0 storage=9.57KiB dangling=0 elapsed=4.630ms
INFO [03-15|21:24:33.568] IPC endpoint opened                      url=/app/node1/geth.ipc
INFO [03-15|21:24:33.569] Generated JWT secret                     path=/app/node1/geth/jwtsecret
INFO [03-15|21:24:33.569] New local node record                    seq=1,678,915,473,568 id=b63b33d1c987399b ip=127.0.0.1 udp=0 tcp=30303
INFO [03-15|21:24:33.569] Started P2P networking                   self="enode://63ccfe4361ebb4e482a1fcddd8737f131b4bb1f33b95fc1b303002bd5e89a066699b944a7f3be65da435c9fa7e5f2a3ae0f3f73ac7fdcf7990fdde90a86616c7@127.0.0.1:30303?discport=0"
INFO [03-15|21:24:33.569] HTTP server started                      endpoint=127.0.0.1:8545 auth=false prefix= cors=* vhosts=localhost
INFO [03-15|21:24:33.569] WebSocket enabled                        url=ws://127.0.0.1:8551
INFO [03-15|21:24:33.569] HTTP server started                      endpoint=127.0.0.1:8551 auth=true  prefix= cors=localhost vhosts=localhost
INFO [03-15|21:24:34.224] Unlocked account                         address=0xd025f1E3f633eB4816661c68155B6bF9cc43BE16
INFO [03-15|21:24:34.225] Transaction pool price threshold updated price=0
INFO [03-15|21:24:34.225] Updated mining threads                   threads=3
INFO [03-15|21:24:34.225] Transaction pool price threshold updated price=1,000,000,000
ERROR[03-15|21:24:34.225] Cannot start mining without etherbase    err="etherbase must be explicitly specified"
Fatal: Failed to start mining: etherbase missing: etherbase must be explicitly specified
root@47e2f8084004:~#
root@47e2f8084004:~# geth version
Geth
Version: 1.11.3-stable
Git Commit: 5ed08c4735c9d034fface08e4c400f1bd1ba8318
Architecture: amd64
Go Version: go1.20.2
Operating System: linux
GOPATH=
GOROOT=
root@47e2f8084004:~#
Basically, the command works, but some minor issue I don't have any experience on this.  I didn't change anything for the json file and the password file.
Please show me a correct command example.
Thanks,

**Comment by zydjohnHotmail** [comment]
Hello:
With some super-human genius help, I figured out, the following is the correct command: # geth --networkid 33 --mine --miner.threads 3 --datadir=/app/node1 --nodiscover --http --http.corsdomain "*" --nat "any" --miner.etherbase 'd025f1e3f633eb4816661c68155b6bf9cc43be16' --unlock 'd025f1e3f633eb4816661c68155b6bf9cc43be16' --password /app/node1/password.txt --allow-insecure-unlock
However, the mining seems endless, how can I stop mining, like in BTC Regtest, I can decide how many blocks to mine. For example, if I want to mine only 100 blocks, how can I re-write the command with some other parameters?
Thanks,

**Comment by marcosmartinez7** [comment]
Glad you managed to fix the issue. 

I dont know if you can mine only 100 blocks. Mining is endless AFAIK, you can just Ctrl + C and finish geth process 

Why do you need that?

**Comment by zydjohnHotmail** [comment]
Hi,
I ignore some warning when running the command to start mining, the warning message is: 
WARN [03-16|09:47:52.414] Post-merge network, but no beacon client seen. Please launch one to follow the chain!
After the warning, the message stops, the following is the message from the warning until the last message:
WARN [03-16|09:47:52.414] Post-merge network, but no beacon client seen. Please launch one to follow the chain!
INFO [03-16|09:47:52.541] Generating DAG in progress               epoch=1 percentage=82 elapsed=15.923s
INFO [03-16|09:47:52.730] Generating DAG in progress               epoch=1 percentage=83 elapsed=16.113s
INFO [03-16|09:47:52.919] Generating DAG in progress               epoch=1 percentage=84 elapsed=16.301s
INFO [03-16|09:47:53.111] Generating DAG in progress               epoch=1 percentage=85 elapsed=16.494s
INFO [03-16|09:47:53.298] Generating DAG in progress               epoch=1 percentage=86 elapsed=16.680s
INFO [03-16|09:47:53.485] Generating DAG in progress               epoch=1 percentage=87 elapsed=16.868s
INFO [03-16|09:47:53.673] Generating DAG in progress               epoch=1 percentage=88 elapsed=17.055s
INFO [03-16|09:47:53.863] Generating DAG in progress               epoch=1 percentage=89 elapsed=17.246s
INFO [03-16|09:47:54.050] Generating DAG in progress               epoch=1 percentage=90 elapsed=17.433s
INFO [03-16|09:47:54.237] Generating DAG in progress               epoch=1 percentage=91 elapsed=17.619s
INFO [03-16|09:47:54.421] Generating DAG in progress               epoch=1 percentage=92 elapsed=17.804s
INFO [03-16|09:47:54.619] Generating DAG in progress               epoch=1 percentage=93 elapsed=18.002s
INFO [03-16|09:47:54.816] Generating DAG in progress               epoch=1 percentage=94 elapsed=18.199s
INFO [03-16|09:47:55.005] Generating DAG in progress               epoch=1 percentage=95 elapsed=18.388s
INFO [03-16|09:47:55.196] Generating DAG in progress               epoch=1 percentage=96 elapsed=18.579s
INFO [03-16|09:47:55.406] Generating DAG in progress               epoch=1 percentage=97 elapsed=18.789s
INFO [03-16|09:47:55.610] Generating DAG in progress               epoch=1 percentage=98 elapsed=18.993s
INFO [03-16|09:47:55.987] Generating DAG in progress               epoch=1 percentage=99 elapsed=19.370s
INFO [03-16|09:47:55.990] Generated ethash verification cache      epoch=1 elapsed=19.372s

The network ID#33 seems not a good one, which network ID I can use, so the local Regtest net can be mined without warning?
Please advise!
Thanks,

**Comment by marcosmartinez7** [comment]
Its ok, 33 is for regrets, it doesn't collide with mainnet and testnet 

the warning shouldn't be preventing you from doing anything, are you experimenting some problem?

**Comment by zydjohnHotmail** [comment]
Hello:
I found the issue seems to be: I have to run a consensus client, while geth is just a execution client; for old version of Ethereum, running geth is enough, but nowadays, since Ethereum has migrated from Proof of Work to Proof of Stake, you have to use both a consensus client and an execution client to make it work.  I have to read more documents on how to use them. 
Do you have experience with consensus client? Like: lighthouse or lodestart?

**Comment by marcosmartinez7** [comment]
> Hello:
> I found the issue seems to be: I have to run a consensus client, while geth is just a execution client; for old version of Ethereum, running geth is enough, but nowadays, since Ethereum has migrated from Proof of Work to Proof of Stake, you have to use both a consensus client and an execution client to make it work. I have to read more documents on how to use them.
> Do you have experience with consensus client? Like: lighthouse or lodestart?

Hi, no sorry. If you find something let me know. Would be nice to improve this repo
